pymeasure/experiment/results.py

```diff
@@ -129,16 +131,54 @@ def __init__(self, columns, delimiter=','):
         """
         super().__init__()
         self.columns = columns
+        self.units = self._parse_columns(columns)
         self.delimiter = delimiter
 
+    @staticmethod
+    def _parse_columns(columns):
+        """Parse the columns to get units in parenthesis."""
+        units_pattern = r"\((?P<units>[\w/\(\)\*\t]+)\)"
+        units = {}
+        for column in columns:
+            try:
+                match = re.search(units_pattern, column)
+            except TypeError:
+                match = None
+
+            if match:
+                m = match.groupdict()
+                if 'units' in m:
+                    units[column] = pint.Unit(m['units'])
```

Comment:
Don't you want to duplicate the unit information in the column key? You end up with `{"accel_voltage (kV)": "kV"}`

Reply:
also, for consisteny, don't you want to assign, e.g. `None` for the entries without a unit? (e.g. to keep the size/length of `CSVFormatter.columns` and `CSVFormatter.units` the same (or actually unify them in one data structure)

Reply:
> Do you want to duplicate the unit information in the column key? You end up with `{"accel_voltage (kV)": "kV"}`

yes, I duplicate the information. Removing 'kV' silently from the column could create confusion addressing it with data. The dictionary with the "kV" entry is for faster lookup of the unit, such that I have to parse just once and not for every data point.



> also, for consisteny, don't you want to assign, e.g. `None` for the entries without a unit? (e.g. to keep the size/length of `CSVFormatter.columns` and `CSVFormatter.units` the same (or actually unify them in one data structure)

It's a dictionary. I know, that a not existing entry means, I did not specify anything. Yes, we could make it into a list of the same size with Nones, but I prefer a dictionary, which contains just the necessary information and not a row of None.

Reply:
I did not say turn it into a list, a dict is the correct choice here. 
> I know, that a not existing entry means, I did not specify anything.

Yes, _you_ know. :-) Someone else might not and will wonder what the deal with the missing column entries in a dict column->unit might mean. "Explicit is better than implicit" and all that.

In the end it's probably a style/taste question. :-)

Reply:
I don't know, whether an explicit None makes it more clearer that you did not specify a unit to check. Basically dict.get() will return None, if something does not exist, such that it is not explicit anymore.
If we want to be explicit, I'd use '-' to indicate no unit.

Reply:
OK. I'll leave the choice to you, whatever fits better into the workflow (also with pint parsing etc).